src/app.js

```diff
@@ -59,7 +59,11 @@ function listen (koaApp, config, ledgers) {
   // Start a coroutine that connects to the backend and
   // subscribes to all the ledgers in the background
   co(function * () {
-    yield backend.connect()
+    const success = yield backend.connect()
+    if (!success) {
+      log('app').error('backend initialization has failed - the connector will now exit')
+      process.exit(1)
```

Comment:
In synchronous code I would agree. However this is part of a coroutine and is asynchronous. The former implementation was actually throwing an exception if a pair wasn't correct. The only effect was to make a promise fail, silently. That was basically the bug. So making sure `connect` is typed makes sense here I think.